### PR TITLE
Switch to conduit for AMGX configuration

### PR DESCRIPTION
**Pull request by joshessman-llnl**
This PR revises #302 - I think when I put that one together I wasn't aware of the extent of JSON support in Conduit, which we're already building as a TPL.

### CODE REVIEW

File: src/serac/physics/utilities/equation_solver.cpp [hunk 1]
```diff
@@ -40,130 +40,28 @@ EquationSolver::EquationSolver(MPI_Comm comm, const LinearSolverOptions& lin_opt
 
 namespace detail {
 #ifdef MFEM_USE_AMGX
-using JSONLiteral  = std::variant<std::string, double, int>;
```
Comment (jamiebramwell) [inline src/serac/physics/utilities/equation_solver.cpp:43]: and another JSON parser bites the dust! 🎉

**Review by jamiebramwell — APPROVED** [review]

File: src/serac/physics/utilities/equation_solver.cpp [hunk 2]
```diff
@@ -40,130 +40,28 @@ EquationSolver::EquationSolver(MPI_Comm comm, const LinearSolverOptions& lin_opt
...
-    sep = ',';
-  }
-  out << "\n" << indent << "}";
-  return out;
-}
-
 std::unique_ptr<mfem::AmgXSolver> configureAMGX(const MPI_Comm comm, const AMGXPrec& options)
 {
   auto              amgx                 = std::make_unique<mfem::AmgXSolver>();
   static const auto default_prec_options = []() {
```
Comment (white238) [inline src/serac/physics/utilities/equation_solver.cpp]: Why are these 2 (was 3) lambdas here?

Comment (joshessman-llnl) [reply src/serac/physics/utilities/equation_solver.cpp]: Copy semantics are different for `conduit::Node`s so it made more sense to just assign the 3 values when in verbose mode

**Review by white238 — APPROVED** [review]

**Comment by joshessman-llnl** [comment]
LGTM

**Pull request merged**